packages/extractor-vue/src/vue-extractor.ts

```diff
@@ -12,13 +12,16 @@ export const vueExtractor: ExtractorType = {
     onMessageExtracted,
     ctx: ExtractorCtx
   ) {
-    const { descriptor } = parse(code, {
+    const parsed = parse(code, {
       sourceMap: true,
       filename,
       ignoreEmpty: true,
     })
+    // this handles older SFC compilers (2.6) which return the descriptor directly
+    // @ts-ignore of course we're using typings for more recent SFC compiler so TS disagrees
+    const descriptor = parsed.descriptor || parsed as SFCDescriptor
 
-    const compiledTemplate = compileTemplate({
+    const compiledTemplate = descriptor.template && compileTemplate({
```

Comment:
`"@vue/compiler-sfc": "^3.2.47"` written as dependency of `@lingui/extractor-vue` don't understand how this extractor would use older version and why.

Reply:
Yeah, me neither. My client uses a monorepo for most of the front-end (which is a couple of Vue 2.6 apps and some libraries), and I don't really understand how lerna and yarn decide what to install (we don't have a forced resolution for those packages) but the result is that when I run `lingui extract`, it ends up using compiler-sfc@2.6 and it crashes without this code.

Reply:
Actually the culprit may be the babel config (which lingui uses implicitly) rather than yarn/lerna.

Reply:
I think it shouldn't be in the library code, because it's rather monorepo configuration issue. You need to check that all packages explicitly states all dependencies and there no "phantom dependecies" - for example `lingui extract` is used in some packages but not directly installed there.

Reply:
> Actually the culprit may be the babel config (which lingui uses implicitly) rather than yarn/lerna. 

that is not the case, babel config in extractor is sealed and not affected by environment configuration.

Reply:
lingui and related packages are all a direct, explicit dependency of the app I'm using it for.

I agree that this shouldn't happen, but I didn't hallucinate the stack trace.

I don't mind removing that change. I can't use this package as-is anyway because of how the base extractor is imported (another issue with babel, maybe). But please consider that it's just a few more bytes and it may help others with messed-up configs.

Reply:
>  because of how the base extractor is imported (another issue with babel, maybe).

Could you elaborate a bit on that?

Reply:
Like I said in the other discussion, when this is transpiled to commonjs, the result is `const babel = require('@lingui/cli/api/extractors/babel');` but what actually works (for me) is `const babel = require('@lingui/cli/api/extractors/babel').default;`.

Reply:
well, i know why this happend. I will try to fix that. 

Regarding `const descriptor = parsed.descriptor || parsed as SFCDescriptor` i would suggest removing it because it brings maintenance burden (we are not responsible for misconfigured monorepos whatsoever) and this would be good to be merged

Reply:
OK. I removed that commit.